src/ensembl/src/content/app/browser/track-panel/trackPanelState.ts

```diff
@@ -1,11 +1,32 @@
-import { TrackSet } from './trackPanelConfig';
+import { TrackSet, GenomeTrackStates } from './trackPanelConfig';
 import browserStorageService from '../browser-storage-service';
+import trackPanelStorageService from './track-panel-storage-service';
+
+export type Bookmark = {
+  genome_id: string;
+  object_id: string;
+  object_type: string;
+  location: {
+    chromosome: string;
+    start: number;
+    end: number;
+  };
+  label: string;
+  trackStates: GenomeTrackStates;
```

Comment:
If this is describing data for a previously viewed item, does it need anything at all other than just one single data point: ensembl object id?

Reply:
@azangru as per the Gherkin syntax mentioned in the ticket https://www.ebi.ac.uk/panda/jira/browse/ENSWBSITES-237, we will have to store the track states along with the previously viewed links. This has already been approved by Andrea.